lib/slapi/client/base.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+# Base Client for Slapi Class
+# Its main functions are to:
+#  1. Start the Bot
+#     - Set listeners and which helpers to utilize
+class Slapi
+  def self.run
+    @client.on :hello do
+      @logger.info("Slapi: Successfully connected, welcome '#{@client.self.name}' to the '#{@client.team.name}' team at https://#{@client.team.domain}.slack.com.")
+    end
+
+    @client.on :message do |data|
+      case data.text
+      when /#{bot_prefix(data)}ping/ then
+        @logger.debug("Slapi: #{data.user} requested ping")
+        ping(data)
+      when /#{bot_prefix(data)}help/ then
+        @logger.debug("Slapi: #{data.user} requested help")
+        get_help(data)
+      when /#{bot_prefix(data)}reload/ then
+        @logger.debug("Slapi: #{data.user} requested plugin reload")
+        reload(data)
+      when /#{bot_prefix(data)}/ then
+        @logger.debug("Slapi: #{data.user} request forwarded to check against plugins")
+        plugin(data)
+      end unless data.user == @client.self.id
+    end
+    @client.start_async
+  end
+
+  def self.ping(data)
+    chat(
+      data,
+      title: 'Bot Check',
+      text: 'PONG',
+      color: '#229954'
```

Comment:
Yea, there should only be three colors. Would could possibly set constants for this as well.